python/ray/serve/scripts.py

```diff
@@ -431,12 +453,19 @@ def run(
         )
 
     http_options = {"host": host, "port": port, "location": "EveryNode"}
-    # Merge http_options with the ones on ServeDeploySchema. If host and/or port is
-    # passed by cli, those continue to take the priority
+    grpc_options = gRPCOptions()
+    # Merge http_options and grpc_options with the ones on ServeDeploySchema. If host
+    # and/or port is passed by cli, those continue to take the priority
     if is_config and isinstance(config, ServeDeploySchema):
         config_http_options = config.http_options.dict()
         http_options = {**config_http_options, **http_options}
-    client = _private_api.serve_start(detached=True, http_options=http_options)
+        grpc_options = gRPCOptions(**config.grpc_options.dict())
```

Comment:
Yes, this is for brining the config from the serve config in when start serve. Without this the serve options on the config will not take affect.

Reply:
Why can't we just do 

```suggestion
        grpc_options = config.grpc_options
```

Reply:
good point, but coming from the serve config file this would be an `gRPCOptionsSchema` object which is different than a `gRPCOptions` and still need to do type casing 😅